src/radial.sol

```diff
@@ -81,22 +81,32 @@ contract Radial {
         totalSupply    = add(totalSupply, wad);
         emit Transfer(address(0), usr, wad);
     }
-    function burn(address usr, uint wad) external {
+
+    function _burn(address sender, address usr, uint wad) internal {
         require(balanceOf[usr] >= wad, "cent/insufficient-balance");
-        if (usr != msg.sender && allowance[usr][msg.sender] != uint(-1)) {
-            require(allowance[usr][msg.sender] >= wad, "cent/insufficient-allowance");
-            allowance[usr][msg.sender] = sub(allowance[usr][msg.sender], wad);
+        if (usr != sender && allowance[usr][sender] != uint(-1)) {
+            require(allowance[usr][sender] >= wad, "cent/insufficient-allowance");
+            allowance[usr][sender] = sub(allowance[usr][sender], wad);
         }
         balanceOf[usr] = sub(balanceOf[usr], wad);
         totalSupply    = sub(totalSupply, wad);
         emit Transfer(usr, address(0), wad);
     }
+
+    function burn(address usr, uint wad) external {
+        _burn(msg.sender, usr, wad);
+    }
+
     function approve(address usr, uint wad) external returns (bool) {
         allowance[msg.sender][usr] = wad;
         emit Approval(msg.sender, usr, wad);
         return true;
     }
+
     // --- Alias ---
+    function burnFrom(address usr, uint wad) external {
+        _burn(msg.sender, usr, wad);
```

Comment:
Given both `burnFrom` and `burn` call `_burn` I think we should not add the internal method `_burn` and just do this:
```suggestion
        burn(usr, wad);
```

Reply:
The reason why I added the internal function, was bc for some reason the inner `burn` call was not passing the sender along with it. Do you have any ideas?